src/scripts/exclusion_term_expansion.py

```diff
@@ -0,0 +1,422 @@
+        #  - ordo
+        df = pd.read_csv(results_path).fillna('')
+    except pd.errors.EmptyDataError:
+        # remove: so that it doesn't read this from cache, though it could be that there were really no results.
+        os.remove(results_path)
+        # could also do `except pd.errors.EmptyDataError as err`, `raise err` or give option for this as a param to func
+        df = pd.DataFrame()  # could also return None
+
+    return df
+
+
+def expand_ontological_exclusions(
+    onto_path: str, exclusions_df: pd.DataFrame, prefix_sparql_strings: List[str] = None, cache_suffix: str = None,
+    export_fields=['term_id', 'exclusion_reason'], uri_fields=['term_id', 'child_id']
+) -> pd.DataFrame:
+    """Using an ontology file and an exclusions file, query and get return an extensional list of exclusions.
+    cache_suffix: Used by and explained in sparql_jinja2_robot_query().
+    prefix_sparql_strings: of the form 'prefix [PREFIX]: <[URI]>"""
+    # Variable names: kids0=Children excluded; kids1=Children included
+    df_kids1 = exclusions_df[exclusions_df['exclude_children'] == False]
+    df_kids0 = exclusions_df[exclusions_df['exclude_children'] == True]
+
+    # Terms w/ non-excluded children
+    if len(df_kids1) > 0:
+        terms_kids1: List[str] = list(set(df_kids1['term_id']))  # QA: set() removes any duplicates
+        df_kids1_results_1st: pd.DataFrame = sparql_jinja2_robot_query(
+            prefixes=prefix_sparql_strings,
+            terms=terms_kids1,
+            query_template_path=SPARQL_CHILD_INCLUSION_PATH,
+            onto_path=onto_path,
+            cache_suffix=cache_suffix)
+
+        # Massage results
+        # # Convert URI back to prefix
+        for fld in uri_fields:
+            df_kids1_results_1st[fld] = df_kids1_results_1st[fld].apply(uri_to_curie)
+        # # JOIN: To get exclusion_reason
+        df_kids1_results = pd.merge(df_kids1_results_1st, exclusions_df, how='left', on='term_id').fillna('')
+        # # Drop parent term data
+        df_kids1_results = df_kids1_results[['child_id', 'exclusion_reason']]
+        # # Rename child fields as they are now top-level terms
+        df_kids1_results = df_kids1_results.rename(columns={'child_id': 'term_id'})
+    else:
+        df_kids1_results = df_kids1  # empty, but columns needed for concat below
+
+    # Terms w/ excluded children
+    pass  # the list itself is all we need; no query necessary
+
+    # Massage
+    # - Drop unneeded fields
+    df_kids0 = df_kids0[export_fields]
+    df_kids1_results = df_kids1_results[export_fields]
+
+    # CONCAT: Combine term child results w/ terms where children were excluded
+    df_results = pd.concat([df_kids0, df_kids1_results]).fillna('')
+
+    return df_results
+
+
+# TODO: pass prefixes to this func. need to move logic up to run() I think and pass down to both funcs
+# TODO:
+def get_non_inclusions(mirror_signature_path: str, component_signature_path: str, prefix_uris: List[str]) -> pd.DataFrame:
```

Comment:
@nicolevasilevsky @sabrinatoro @matentzn Per Nico's instruction from [the google doc](https://docs.google.com/document/d/1-bFd6GeHcLHImjKImML_j0uAyKENIlb5NN64oGP4rwY/edit?disco=AAAAc-gOiSs), I added the `exclusion_reason` `MONDO:excludeNonDisease` for all of these exclusions (in 'mirror' but not in 'component'). However, I noticed that this reason is not listed in the docs: https://mondo.readthedocs.io/en/latest/editors-guide/exclusion-reasons/

Reply:
I see an issue has been made for this. Marking resolved!